soc/arm/microchip_mec/Kconfig

```diff
@@ -11,6 +12,212 @@ config SOC_FAMILY
+
+choice MCHP_MEC_HEADER_FLASH_SIZE_CHOICE
+	prompt "Flash size"
+	default MCHP_MEC_HEADER_FLASH_SIZE_16M
+	help
+	  This sets the SPI flash size.
+
+config MCHP_MEC_HEADER_FLASH_SIZE_256K
+	bool "SPI flash size 256K Bytes"
+	help
+	  The SPI flash size is 256K Bytes.
+
+config MCHP_MEC_HEADER_FLASH_SIZE_512K
+	bool "SPI flash size 512K Bytes"
+	help
+	  The SPI flash size is 512K Bytes.
+
+config MCHP_MEC_HEADER_FLASH_SIZE_1M
+	bool "SPI flash size 1M Bytes"
+	help
+	  The SPI flash size is 1M Bytes.
+
+config MCHP_MEC_HEADER_FLASH_SIZE_2M
+	bool "SPI flash size 2M Bytes"
+	help
+	  The SPI flash size is 2M Bytes.
+
+config MCHP_MEC_HEADER_FLASH_SIZE_4M
+	bool "SPI flash size 4M Bytes"
+	help
+	  The SPI flash size is 4M Bytes.
+
+config MCHP_MEC_HEADER_FLASH_SIZE_8M
+	bool "SPI flash size 8M Bytes"
+	help
+	  The SPI flash size is 8M Bytes.
+
+config MCHP_MEC_HEADER_FLASH_SIZE_16M
+	bool "SPI flash size 16M Bytes"
+	help
+	  The SPI flash size is 16M Bytes.
+endchoice
+
+config MCHP_MEC_HEADER_FLASH_SIZE
+	int
+	default 256 if MCHP_MEC_HEADER_FLASH_SIZE_256K
+	default 512 if MCHP_MEC_HEADER_FLASH_SIZE_512K
+	default 1024 if MCHP_MEC_HEADER_FLASH_SIZE_1M
+	default 2048 if MCHP_MEC_HEADER_FLASH_SIZE_2M
+	default 4096 if MCHP_MEC_HEADER_FLASH_SIZE_4M
+	default 8192 if MCHP_MEC_HEADER_FLASH_SIZE_8M
+	default 16384 if MCHP_MEC_HEADER_FLASH_SIZE_16M
+
+choice MCHP_MEC_HEADER_SPI_DRVSTR_CHOICE
+	prompt "Flash drive strength"
+	default MCHP_MEC_HEADER_SPI_DRVSTR_1X
+	help
+	  This sets the SPI flash size.
+
+config MCHP_MEC_HEADER_SPI_DRVSTR_1X
+	bool "SPI flash drive strength multiplier 1"
+	help
+	  The SPI flash size is 256K Bytes.
+
+config MCHP_MEC_HEADER_SPI_DRVSTR_2X
+	bool "SPI flash drive strength multiplier 2"
+	help
+	  The SPI flash size is 256K Bytes.
+
+config MCHP_MEC_HEADER_SPI_DRVSTR_4X
+	bool "SPI flash drive strength multiplier 4"
+	help
+	  The SPI flash size is 512K Bytes.
+
+config MCHP_MEC_HEADER_SPI_DRVSTR_6X
+	bool "SPI flash drive strength multiplier 6"
+	help
+	  The SPI flash size is 1M Bytes.
+endchoice
+
+config MCHP_MEC_HEADER_SPI_DRVSTR
+	string
+	default "1x" if MCHP_MEC_HEADER_SPI_DRVSTR_1X
+	default "2x" if MCHP_MEC_HEADER_SPI_DRVSTR_2X
+	default "4x" if MCHP_MEC_HEADER_SPI_DRVSTR_4X
+	default "6x" if MCHP_MEC_HEADER_SPI_DRVSTR_6X
+
+choice MCHP_MEC_HEADER_SPI_SLEW_RATE_CHOICE
+	prompt "Slew rate of SPI pins"
+	default MCHP_MEC_HEADER_SPI_SLEW_RATE_SLOW
+	help
+	  This sets the slew rate of the SPI pins. Default is slow
+	  slew rate which is 1/2 the AHB clock rate. Fast slew is the
+	  AHB clock rate.
+
+config MCHP_MEC_HEADER_SPI_SLEW_RATE_SLOW
+	bool "SPI pins slew rate is 1/2 AHB frequency"
+
+config MCHP_MEC_HEADER_SPI_SLEW_RATE_FAST
+	bool "SPI pins slew rate is 1x AHB frequency"
+endchoice
+
+config MCHP_MEC_HEADER_SPI_SLEW_RATE
+	string
+	default "slow" if MCHP_MEC_HEADER_SPI_SLEW_RATE_SLOW
+	default "fast" if MCHP_MEC_HEADER_SPI_SLEW_RATE_FAST
+
+config MCHP_MEC_HEADER_FLASH_SPI_MODE
+	int
```

Comment:
Does this need a description? It can't be set to anything otherwise.

Reply:
added prompt